analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CheckArgumentException.cs

```diff
@@ -171,7 +171,7 @@ void ComplexCase(int a)
         void Bar2(int a)
         {
             // See https://github.com/SonarSource/sonar-dotnet/issues/1867
-            throw new ArgumentNullException(null, string.Empty); // Noncompliant
+            throw new ArgumentNullException(null, string.Empty); // rare FN introduced by fixing https://github.com/SonarSource/sonar-dotnet/issues/4180
```

Comment:
Please add two more test cases.
```suggestion
            throw new ArgumentNullException(null, string.Empty); // rare FN introduced by fixing https://github.com/SonarSource/sonar-dotnet/issues/4180
            throw new ArgumentNullException(string.Empty, string.Empty); // rare FN introduced by fixing https://github.com/SonarSource/sonar-dotnet/issues/4180
            throw new ArgumentNullException("      ", string.Empty); // Noncompliant
```

Reply:
nice catch. actually, having `throw new ArgumentNullException("", "")` and forgetting to update it would be quite a common "forget" scenario... so that's a pretty bad FN...

Reply:
and I read better the issue description: `"when the referenced argument type is not resolvable"`

that means that `nameof` is handled ok when the type is solvable. This is really an edge-case